cmd/outline-ss-server/config.go

```diff
@@ -22,71 +22,128 @@ import (
 )
 
 type ServiceConfig struct {
-	Listeners []ListenerConfig
-	Keys      []KeyConfig
-	Dialer    DialerConfig
+	Listeners []ListenerConfig `yaml:"listeners"`
+	Keys      []KeyConfig      `yaml:"keys"`
+	Dialer    DialerConfig     `yaml:"dialer"`
 }
 
 type ListenerType string
 
-const listenerTypeTCP ListenerType = "tcp"
+const (
+	listenerTypeTCP ListenerType = "tcp"
 
-const listenerTypeUDP ListenerType = "udp"
+	listenerTypeUDP             ListenerType = "udp"
+	listenerTypeWebsocketStream ListenerType = "websocket-stream"
+	listenerTypeWebsocketPacket ListenerType = "websocket-packet"
+)
+
+type WebServerConfig struct {
```

Comment:
I'm assuming these are always HTTP?
Perhaps we should add a comment that this should probably listen on localhost addresses, since it's not safe to expose HTTP publicly